Signal/src/util/OWSBackupExportJob.m

```diff
@@ -211,310 +402,297 @@ - (void)configureExport:(OWSBackupJobBoolCompletion)completion
+    if (aborted || self.isComplete) {
         return NO;
     }
-    // TODO: Should we do a database checkpoint?
 
-    DDLogInfo(@"%@ copiedThreads: %llu", self.logTag, copiedThreads);
-    DDLogInfo(@"%@ copiedMessages: %llu", self.logTag, copiedInteractions);
-    DDLogInfo(@"%@ copiedEntities: %llu", self.logTag, copiedEntities);
-    DDLogInfo(@"%@ copiedAttachments: %llu", self.logTag, copiedAttachments);
-    DDLogInfo(@"%@ copiedMigrations: %llu", self.logTag, copiedMigrations);
+    @autoreleasepool {
+        if (![exportStream flush]) {
+            OWSProdLogAndFail(@"%@ Could not flush database snapshots.", self.logTag);
+            return NO;
+        }
+    }
 
-    [self.backupStorage logFileSizes];
+    self.unsavedDatabaseItems = [exportStream.exportItems mutableCopy];
 
-    // Capture the list of files to save.
-    self.databaseFilePaths = [@[
-        self.backupStorage.databaseFilePath,
-        self.backupStorage.databaseFilePath_WAL,
-        self.backupStorage.databaseFilePath_SHM,
-    ] mutableCopy];
+    // TODO: Should we do a database checkpoint?
 
-    // Close the database.
-    tempDBConnection = nil;
-    self.backupStorage = nil;
+    DDLogInfo(@"%@ copiedThreads: %zd", self.logTag, copiedThreads);
+    DDLogInfo(@"%@ copiedMessages: %zd", self.logTag, copiedInteractions);
+    DDLogInfo(@"%@ copiedAttachments: %zd", self.logTag, copiedAttachments);
+    DDLogInfo(@"%@ copiedMigrations: %zd", self.logTag, copiedMigrations);
+    DDLogInfo(@"%@ copiedEntities: %zd", self.logTag, exportStream.totalItemCount);
 
     return YES;
 }
 
-- (void)saveToCloud:(OWSBackupJobCompletion)completion
+- (void)saveToCloudWithCompletion:(OWSBackupJobCompletion)completion
 {
     OWSAssert(completion);
 
     DDLogVerbose(@"%@ %s", self.logTag, __PRETTY_FUNCTION__);
 
-    self.databaseRecordMap = [NSMutableDictionary new];
-    self.attachmentRecordMap = [NSMutableDictionary new];
+    self.savedDatabaseItems = [NSMutableArray new];
+    self.savedAttachmentItems = [NSMutableArray new];
+
+    {
+        unsigned long long totalFileSize = 0;
+        for (OWSBackupExportItem *item in self.unsavedDatabaseItems) {
+            totalFileSize += [OWSFileSystem fileSizeOfPath:item.encryptedItem.filePath].unsignedLongLongValue;
+        }
+        DDLogInfo(@"%@ exporting %@: count: %zd, bytes: %llu.",
+            self.logTag,
+            @"database items",
+            self.unsavedDatabaseItems.count,
+            totalFileSize);
+    }
+    {
+        unsigned long long totalFileSize = 0;
+        for (OWSAttachmentExport *attachmentExport in self.unsavedAttachmentExports) {
+            totalFileSize += [OWSFileSystem fileSizeOfPath:attachmentExport.attachmentFilePath].unsignedLongLongValue;
+        }
+        DDLogInfo(@"%@ exporting %@: count: %zd, bytes: %llu.",
+            self.logTag,
+            @"attachment items",
+            self.unsavedAttachmentExports.count,
+            totalFileSize);
```

Comment:
Could you either only log the attachment size here, or clarify the log to indicate this is the db+attachment size?

Reply:
Done.